metadatastore/commands.py

```diff
@@ -39,167 +99,524 @@ def inner(*args, **kwargs):
+
+    Parameters
+    ----------
+    descriptor : doc.Document or dict or str
+        The EventDescriptors to get the Events for.  Can be either
+        a Document/dict with a 'uid' key or a uid string
+
+    Yields
+    ------
+    event : doc.Document
+        All events for the given EventDescriptor from oldest to
+        newest
+    """
+    descriptor_uid = doc_or_uid_to_uid(descriptor)
+    descriptor = descriptor_given_uid(descriptor_uid)
+
+    oid = _DESCRIPTOR_UID_to_OID_MAP[descriptor_uid]
+
+    col = Event._get_collection()
+    ev_cur = col.find({'descriptor_id': oid},
+                      sort=[('time', 1)])
+
+    for ev in ev_cur:
+        # ditch the ObjectID
+        del ev['_id']
+        # pop the descriptor oid
+        del ev['descriptor_id']
+        # replace it with the defererenced descriptor
+        ev['descriptor'] = descriptor
+        data = ev.pop('data')
+        ev['timestamps'] = {k: v[1] for k, v in data.items()}
+        ev['data'] = {k: v[0] for k, v in data.items()}
+
+        # wrap it in our fancy dict
+        ev = doc.Document('Event', ev)
+
+        yield ev
+
+
+def _transpose(in_data, keys, field):
+    out = {k: [None] * len(in_data) for k in keys}
+    for j, ev in enumerate(in_data):
+        dd = ev[field]
+        for k in keys:
+            out[k][j] = dd[k]
+    return out
+
+
+@_ensure_connection
+def fetch_events_table(descriptor):
+    """All event data as tables
+
+    Parameters
+    ----------
+    descriptor : dict or str
+        The EventDestriptor to get the events for.  Can be either
+        a dict or a uid.
+
+    Returns
+    -------
+    descriptor : doc.Document
+        EventDescriptor document
+
+    data_table : dict
+        dict of lists of the transposed data
+
+    uids : list
+        The uids of each of the events.
+
+    seq_nums : list
+        The sequence number of each event.
+
+    times : list
+        The time of each event.
+
+    uids : list
+        The uid of each event.
+
+    timestamps_table : dict
+        The timestamps of each of the measurements as dict of lists.  Same
+        keys as `data_table`.
+    """
+    desc_uid = doc_or_uid_to_uid(descriptor)
+    descriptor = descriptor_given_uid(desc_uid)
+    # this will get more complicated once transpose caching layer is in place
+    all_events = list(fetch_events_generator(desc_uid))
+
+    # get event sequence numbers
+    seq_nums = [ev['seq_num'] for ev in all_events]
+
+    # get event times
+    times = [ev['time'] for ev in all_events]
+
+    # get uids
+    uids = [ev['uid'] for ev in all_events]
+
+    keys = list(descriptor['data_keys'])
+
+    # TODO make sure this is fast
+    # get data values
+    data_table = _transpose(all_events, keys, 'data')
+
+    # get timestamps
+    timestamps_table = _transpose(all_events, keys, 'timestamps')
+
+    # return the whole lot
+    return descriptor, data_table, seq_nums, times, uids, timestamps_table
+
+
+def db_disconnect():
```

Comment:
could you move these two functions up to where `_ensure_connection` is defined? They seem rather out of place in the file here